qutebrowser/browser/webkit/webkithistory.py

```diff
@@ -48,7 +50,8 @@ def historyContains(self, url_string):
         Return:
             True if the url is in the history, False otherwise.
         """
-        return url_string in self._history.history_dict
+        with debug.log_time('sql', 'historyContains'):
+            return url_string in self._history
```

Comment:
Before we stored the entire history in RAM for this, so that's already quite an improvement over the status quo.

And how many URLs are you really going to visit in a single browsing session? Trying this out with the first 5000 items in my history (which already seems like a lot), I get a set which takes 430kB of RAM. That's a price I'll gladly pay if it makes things noticably faster, and nothing compared to the RAM QtWeb{Kit,Engine} use.